xchange-core/src/main/java/org/knowm/xchange/service/marketdata/MarketDataService.java

```diff
@@ -74,6 +94,26 @@ default OrderBook getOrderBook(CurrencyPair currencyPair, Object... args) throws
     throw new NotYetImplementedForExchangeException("getOrderBook");
   }
 
+  /**
+   * Get an order book representing the current offered exchange rates (market depth)
+   *
+   * @param args Optional arguments. Exchange-specific
+   * @return The OrderBook, null if some sort of error occurred. Implementers should log the error.
+   * @throws ExchangeException - Indication that the exchange reported some kind of error with the
+   *     request or response
+   * @throws NotAvailableFromExchangeException - Indication that the exchange does not support the
+   *     requested function or data
+   * @throws NotYetImplementedForExchangeException - Indication that the exchange supports the
+   *     requested function or data, but it has not yet been implemented
+   * @throws IOException - Indication that a networking error occurred while fetching JSON data
+   */
+  default OrderBook getOrderBook(Instrument instrument, Object... args) throws IOException {
+    if(instrument instanceof CurrencyPair) {
```

Comment:
again same as above. lets make the `Instrument` and `CurrencyPair` methods separate so we can decommission the `CurrencyPair` methods at some point.